lib/instance.js

```diff
@@ -435,6 +501,22 @@ instance.prototype.activate_module = function (id, modin) {
 	try {
 		self.active[id] = new mod(self.system, id, Object.assign({}, self.store.db[id]))
 
+		if (mod.GetUpgradeScripts === undefined) {
+			if (self.active[id]._versionscripts !== undefined && self.active[id]._versionscripts.length > 0) {
+				// New instances do not need to be upgraded
+				self.store.db[id]._configIdx = self.active[id]._versionscripts.length - 1
+			}
+		} else {
+			// Module isn't using new flow of upgrade scripts, so make sure it isnt using the old flow
+			self.active[id].disableInstanceUpgradeScripts()
+
+			const scripts = mod.GetUpgradeScripts()
+			if (scripts !== undefined && scripts.length > 0) {
+				// New instances do not need to be upgraded
+				self.store.db[id]._configIdx = scripts.length - 1
+			}
+		}
+
 		if (self.active[id]._versionscripts !== undefined && self.active[id]._versionscripts.length > 0) {
```

Comment:
We may have an existing bug here.  This `if`'s internal says `// New instances do not need to be upgraded` but `activate_module` is also called by `instance_enable` and if an instance was disabled on startup and later enabled it would present the same and have its `_configIdx` overwritten by this block?

Reply:
yep, looks like you are right there. I have a fix for that now